compiler/rustc_codegen_cranelift/src/abi/mod.rs

```diff
@@ -916,25 +916,26 @@ pub(crate) fn codegen_call_with_unwind_action(
 pub(crate) fn lib_call_arg_param(tcx: TyCtxt<'_>, ty: Type, is_signed: bool) -> AbiParam {
     let param = AbiParam::new(ty);
     if ty.is_int() && u64::from(ty.bits()) < tcx.data_layout.pointer_size().bits() {
-        match (&tcx.sess.target.arch, tcx.sess.target.vendor.as_ref()) {
-            (Arch::X86_64, _) | (Arch::AArch64, "apple") => match (ty, is_signed) {
+        match &tcx.sess.target.arch {
+            Arch::X86_64 | Arch::AArch64 if tcx.sess.target.is_like_darwin => match (ty, is_signed)
```

Comment:
Nit: Add parentheses, it's not clear to me that this won't change behaviour otherwise?

```suggestion
            Arch::X86_64 | (Arch::AArch64 if tcx.sess.target.is_like_darwin) => match (ty, is_signed)
```